contrib/babelfishpg_tds/src/backend/tds/tdstypeio.c

```diff
@@ -1448,6 +1451,56 @@ TdsTypeUIDToDatum(StringInfo buf)
 	PG_RETURN_POINTER(uuid);
 }
 
+/* Helper Function to convert Spatial Type values into Datum. */
+Datum
+TdsTypeSpatialToDatum(StringInfo buf)
+{
+	bytea	   *result;
+	int		   nbytes,
+			   npoints;
+	StringInfo  destBuf = makeStringInfo();
+
+	/*
+	 * Here the incoming buf format is -> 4 Byte SRID + 2 Byte Geometry Type + (16 Bytes)*npoints
+	 * But Driver expects -> 4 Byte SRID + 4 Byte Type + 4 Byte npoints + (16 Bytes)*npoints
+	 */
+	/* We are copying first 4 Byte SRID from buf */
+	appendBinaryStringInfo(destBuf, buf->data + buf->cursor, 4);
+	
+	npoints = (buf->len - buf->cursor - 6)/16;
+	nbytes = buf->len - buf->cursor + 6;
+	result = (bytea *) palloc0(nbytes + VARHDRSZ);
+	SET_VARSIZE(result, nbytes + VARHDRSZ);
+
+	/* Here we are handling the 8 bytes (4 Byte Type + 4 Byte npoints) which driver expects for 2-D point */
+	if (buf->data[buf->cursor + 4] == 1 && buf->data[buf->cursor + 5] == 12)
+	{
+		appendStringInfoChar(destBuf, POINTTYPE & 0xFF);
+		appendStringInfoChar(destBuf, (POINTTYPE >> 8)  & 0xFF);
+		appendStringInfoChar(destBuf, (POINTTYPE >> 16)  & 0xFF);
+		appendStringInfoChar(destBuf, (POINTTYPE >> 24)  & 0xFF);
+
+		appendStringInfoChar(destBuf, npoints & 0xFF);
+		appendStringInfoChar(destBuf, (npoints >> 8)  & 0xFF);
+		appendStringInfoChar(destBuf, (npoints >> 16)  & 0xFF);
+		appendStringInfoChar(destBuf, (npoints >> 24)  & 0xFF);
```

Comment:
We can append a integer value like [this ](https://github.com/babelfish-for-postgresql/postgresql_modified_for_babelfish/blob/BABEL_3_X_DEV__PG_15_X/src/include/libpq/pqformat.h#L75-L81)

Reply:
Done